### ISSUE TITLE
WaveRNN + CUDA 11 + pytorch==1.10.0 == grad norm is NaN? Help needed.

### ISSUE THREAD

**Issue by michael-conrad** [Tomiinek/Multilingual_Text_to_Speech#66]
I'm trying to train a new WaveRNN model to match a Cherokee and other voices setup for Multilingual_Text_to_Speech.

My system requires CUDA 11. I've attached my environment.yml file should that help any.
[environment.yml.zip](https://github.com/Tomiinek/Multilingual_Text_to_Speech/files/7574651/environment.yml.zip)

I had to change the following to prevent an error about numpy(cpu) vs tensor(gpu):

```python
diff --git a/scripts/train_wavernn.py b/scripts/train_wavernn.py
index 144b306..4d9200e 100644
--- a/scripts/train_wavernn.py
+++ b/scripts/train_wavernn.py
@@ -52,7 +52,7 @@ def voc_train_loop(paths: Paths, model: WaveRNN, loss_func, optimizer, scheduler
             loss.backward()
             if hp.voc_clip_grad_norm is not None:
                 grad_norm = torch.nn.utils.clip_grad_norm_(model.parameters(), hp.voc_clip_grad_norm)
-                if np.isnan(grad_norm):
+                if torch.isnan(grad_norm).any():
                     print('grad_norm was NaN!')
             optimizer.step()
```

And now I'm getting grad norm is NaN. After the first iteration, loss goes to NaN.

```python
Initialising Model...

Trainable Parameters: 4.744M
Saving to existing latest checkpoint...
Saving latest weights: /home/muksihs/git/cherokee-wavernn/WaveRNN/checkpoints/chr_en_comvoi.wavernn/latest_weights.pyt
Saving latest optimizer state: /home/muksihs/git/cherokee-wavernn/WaveRNN/checkpoints/chr_en_comvoi.wavernn/latest_optim.pyt
grad_norm was NaN!
| Epoch: 1/1981 (1/505) | Loss: 6.9322 | LR: 0.0010 | 2.3 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (2/505) | Loss: nan | LR: 0.0010 | 2.9 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (3/505) | Loss: nan | LR: 0.0010 | 3.2 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (4/505) | Loss: nan | LR: 0.0010 | 3.4 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (5/505) | Loss: nan | LR: 0.0010 | 3.5 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (6/505) | Loss: nan | LR: 0.0010 | 3.6 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (7/505) | Loss: nan | LR: 0.0010 | 3.6 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (8/505) | Loss: nan | LR: 0.0010 | 3.7 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (9/505) | Loss: nan | LR: 0.0010 | 3.7 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (10/505) | Loss: nan | LR: 0.0010 | 3.8 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (11/505) | Loss: nan | LR: 0.0010 | 3.8 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (12/505) | Loss: nan | LR: 0.0010 | 3.8 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (13/505) | Loss: nan | LR: 0.0010 | 3.8 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (14/505) | Loss: nan | LR: 0.0010 | 3.8 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (15/505) | Loss: nan | LR: 0.0010 | 3.8 steps/s | Step: 0k | grad_norm was NaN!
| Epoch: 1/1981 (16/505) | Loss: nan | LR: 0.0010 | 3.9 steps/s | Step: 0k | grad_norm was NaN!
```

**Comment by Tomiinek** [comment]
I doubt that the change is the problem.
Have you found our what is going on (check your data ...)?

**Comment by michael-conrad** [comment]
I don't know what the issue might be at the moment.

I'm using comvoi plus cherokee audio, I created the spectrograms, then ran preprocess, then ran training.
